In [4]:
 import requests
from datetime import datetime
api_key = '87d845b0b6cf29baa1a73cc34b067a95'
location = input("Enter the city name: ")
 
 
complete_api_link = "https://api.openweathermap.org/data/2.5/weather?q="+location+"&appid="+api_key
api_link = requests.get(complete_api_link)
api_data = api_link.json()
 
temp_city = ((api_data['main']['temp']) - 273.15)
weather_desc = api_data['weather'][0]['description']
hmdt = api_data['main']['humidity']
wind_spd = api_data['wind']['speed']
date_time = datetime.now().strftime("%d %b %Y | %I:%M:%S %p")
 
print ("-------------------------------------------------------------")
print ("Weather Stats for - {}  || {}".format(location.upper(), date_time))
print ("-------------------------------------------------------------")
 
 
print ("Current temperature is: {:.2f} deg C".format(temp_city))
print ("Current weather desc  :",weather_desc)
print ("Current Humidity      :",hmdt, '%')
print ("Current wind speed    :",wind_spd ,'kmph')

Enter the city name: kurnool
-------------------------------------------------------------
Weather Stats for - KURNOOL  || 21 Jun 2021 | 06:25:33 PM
-------------------------------------------------------------
Current temperature is: 31.44 deg C
Current weather desc  : overcast clouds
Current Humidity      : 43 %
Current wind speed    : 8.04 kmph


Name of the city : Kurnool
weather stats for kurnool  || 21 Jun 2021 | 06:25:33 PM
Current temperature : 31.44 deg C
Current weather desc: overcast clouds
Current Humidity    : 43%
Current wind speed  : 8.04 kmph
